lazy_tensor_core/lazy_tensor_core/core/extract_compiled_graph.py

```diff
@@ -90,9 +91,29 @@ def tolazydevice(dev):
             return torch.device("lazy", index=dev.index)
         return dev
 
+    def hasDeviceArg(args, kwargs):
+        return any(isinstance(arg, torch.device) for arg in itertools.chain(args, kwargs.values()))
+
     for nd in model.graph.nodes:
         nd.args = tuple(tolazydevice(arg) for arg in nd.args)
         nd.kwargs = {k: tolazydevice(v) for k, v in nd.kwargs.items()}
+
+        # For torchbench like yolov3, hf_Bart, dynamo generates Fx graph that return
+        # eager tensors on the default device
+        # (check https://gist.github.com/shunting314/eabdf6c769c59bc384469717b8f9bb7f for yolove,
+        # and https://gist.github.com/shunting314/8d5e2d9348a3258959d3954186c48814 for hf_Bart).
+        # To force those tensors on the lazy device, we can not simply override
+        # the device argument since there is no explicit device argument.
+        # What we are doing here is, for the list of covered tensor factory methods
+        # we add a lazy device argument explicity.
+        #
+        # TODO: This solution is no ideal since we may miss some factory methods. In future
+        # when we support lazy mode, this method can be replaced by that.
+        if nd.target in (torch.tensor, torch.full, torch.arange) and not hasDeviceArg(nd.args, nd.kwargs):
```

Comment:
@ezyang @zou3519 is there a way to check if a function is a factory c-tor?

Reply:
cc @anjali411 -- Can we add tags to native_functions.yaml now and query for "all operators that have a particular tag"?

There's a slight mismatch in that the tags in native_functions.yaml are for aten operators and here it looks like we are looking for PyTorch frontend APIs that are factory functions. Though our aten factory functions should map 1:1 with the PyTorch frontend factory functions with the exception of torch.tensor